group-coordinator/src/main/java/org/apache/kafka/coordinator/group/GroupMetadataManager.java

```diff
@@ -1127,7 +1176,9 @@ public void replay(
         Set<String> oldSubscribedTopicNames = new HashSet<>(consumerGroup.subscribedTopicNames());
 
         if (value != null) {
-            ConsumerGroupMember oldMember = consumerGroup.getOrMaybeCreateMember(memberId, true);
+            ConsumerGroupMember oldMember = value.instanceId() != null ?
+                    consumerGroup.getOrMaybeCreateStaticMember(memberId, value.instanceId(), true) :
+                    consumerGroup.getOrMaybeCreateMember(memberId, true);
             consumerGroup.updateMember(new ConsumerGroupMember.Builder(oldMember)
```

Comment:
In there, we need to update the static id mapping in updateMember and removeMember, I think.

Reply:
yes. Removed